FmpDevicePkg/Library/FmpDependencyLib/FmpDependencyLib.c

```diff
@@ -472,18 +494,18 @@ ValidateImageDepex (
   return FALSE;
 }
 
-
 /**
   Get the size of dependencies. Assume the dependencies is validated before
   calling this function.
 
-  @param[in]   Dependencies    Pointer to the EFI_FIRMWARE_IMAGE_DEP.
+  @param[in]  Dependencies   Pointer to the EFI_FIRMWARE_IMAGE_DEP.
 
   @retval  The size of dependencies.
 
 **/
 UINTN
-GetDepexSize (
+EFIAPI
+GetDependencySize (

```

Comment:
No maxsize parameter is a compromise. If the dependency comes from the EFI_FIRMWARE_IMAGE_DESCRIPTOR, we cannot know the maxsize from anywhere. This API is used in CheckFmpDependency() when checking the dependency from descriptor.
Your concern is correct.  Since the API ValidateDependency can output the depex size and have a maxsize parameter, so I will just remove this API and only use it privately in FmpDependencyCheckLib.